pkg/storage/image.go

```diff
@@ -310,13 +313,107 @@ func isValidHostname(hostname string) bool {
 			strings.Contains(hostname, ":") || hostname == "localhost")
 }
 
+func isReferenceFullyQualified(reposName reference.Named) bool {
+	indexName, _, _ := splitReposName(reposName)
+	return indexName != ""
+}
+
+const (
+	// defaultHostname is the default built-in hostname
+	defaultHostname = "docker.io"
+	// legacyDefaultHostname is automatically converted to DefaultHostname
+	legacyDefaultHostname = "index.docker.io"
+	// defaultRepoPrefix is the prefix used for default repositories in default host
+	defaultRepoPrefix = "library/"
+)
+
+// splitReposName breaks a reposName into an index name and remote name
+func splitReposName(reposName reference.Named) (indexName string, remoteName reference.Named, err error) {
+	var remoteNameStr string
+	indexName, remoteNameStr = distreference.SplitHostname(reposName)
+	if !isValidHostname(indexName) {
+		// This is a Docker Index repos (ex: samalba/hipache or ubuntu)
+		// 'docker.io'
+		indexName = ""
+		remoteName = reposName
+	} else {
+		remoteName, err = withName(remoteNameStr)
+	}
+	return
+}
+
+func validateName(name string) error {
+	if err := validateID(strings.TrimPrefix(name, defaultHostname+"/")); err == nil {
+		return fmt.Errorf("Invalid repository name (%s), cannot specify 64-byte hexadecimal strings", name)
```

Comment:
does this error only get raised if you have a 64-byte hex string?

Reply:
I'd say so if you follow the function call to validateID https://github.com/kubernetes-incubator/cri-o/pull/858/files#diff-4366b8905faf0bbe4a80725e573a3b64R355 ...